internal/resources/tanzupackagerepository/resource_package_repository.go

```diff
@@ -6,39 +6,331 @@ SPDX-License-Identifier: MPL-2.0
+	config := m.(authctx.TanzuContext)
+
+	packageRepositoryName, ok := d.Get(nameKey).(string)
+	if !ok {
+		return diag.Errorf("Unable to read package repository name")
+	}
+
+	packageRepositoryNamespace, ok := d.Get(NamespaceKey).(string)
+	if !ok {
+		return diag.Errorf("Unable to read package repository name")
+	}
+
+	scopedFullnameData := scope.ConstructScope(d, packageRepositoryName, packageRepositoryNamespace)
+
+	if scopedFullnameData == nil {
+		return diag.Errorf("Unable to create Tanzu Mission Control package repository entry; Scope full name is empty")
+	}
+
+	err := config.TMCConnection.ClusterPackageRepositoryService.RepositoryResourceServiceDelete(scopedFullnameData.FullnameCluster)
+	if err != nil && !clienterrors.IsNotFoundError(err) {
+		return diag.FromErr(errors.Wrapf(err, "Unable to delete Tanzu Mission Control cluster package repository entry, name : %s", packageRepositoryName))
+	}
+
+	// d.SetId("") is automatically called assuming delete returns no errors, but
+	// it is added here for explicitness.
+	d.SetId("")
+
+	return diags
+}
+
+func resourcePackageRepositoryInPlaceUpdate(ctx context.Context, d *schema.ResourceData, m interface{}) (diags diag.Diagnostics) {
+	config := m.(authctx.TanzuContext)
+
+	packageRepositoryName, ok := d.Get(nameKey).(string)
+	if !ok {
+		return diag.Errorf("Unable to read package repository name")
+	}
+
+	packageRepositoryNamespace, ok := d.Get(NamespaceKey).(string)
+	if !ok {
+		return diag.Errorf("Unable to read package repository name")
+	}
+
+	scopedFullnameData := scope.ConstructScope(d, packageRepositoryName, packageRepositoryNamespace)
+
+	if scopedFullnameData == nil {
+		return diag.Errorf("Unable to create Tanzu Mission Control package repository entry; Scope full name is empty")
+	}
+
+	// nolint: dogsled
+	_, meta, repoSpec, _, err := retrievePackageRepositoryUIDMetaAndSpecFromServer(config, scopedFullnameData, d)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	var updateAvailable bool
+
+	if updateCheckForMeta(d, meta) {
+		updateAvailable = true
+	}
+
+	if updateCheckForSpec(d, repoSpec) {
+		updateAvailable = true
+	}
+
+	if d.HasChange(disabledKey) {
+		updateAvailable = true
+	}
+
+	if !updateAvailable {
+		return
+	}
+
+	pkgRepoReq := &pkgrepositoryclustermodel.VmwareTanzuManageV1alpha1ClusterNamespaceTanzupackageRepositoryRepositoryRequest{
+		Repository: &pkgrepositoryclustermodel.VmwareTanzuManageV1alpha1ClusterNamespaceTanzupackageRepository{
+			FullName: scopedFullnameData.FullnameCluster,
+			Meta:     meta,
+			Spec:     repoSpec,
+		},
+	}
+
+	_, err = config.TMCConnection.ClusterPackageRepositoryService.RepositoryResourceServiceUpdate(pkgRepoReq)
+	if err != nil {
+		return diag.FromErr(errors.Wrapf(err, "Unable to update Tanzu Mission Control cluster package repository entry, name : %s", packageRepositoryName))
+	}
+
+	if d.HasChange(disabledKey) {
```

Comment:
If the update request is only for disabling the repo, then will the code in L239 get executed? What will be the exact behaviour in that case?

Reply:
correct if the update request is only for disabling the repo then the update for spec and meta shouldn't be called
I've updated it appropriately and tested it accordingly.

Reply:
updated PTLA